File: src/main/java/net/tlalka/imager/data/GeoImage.java
package net.tlalka.imager.data;

import java.io.File;

public class GeoImage {

    private File file;
    private double latitude;
    private double longitude;
    private double distance;
    private double direction;
    private String cardinal;

    public GeoImage(File file, double latitude, double longitude) {
        this.file = file;
        this.latitude = latitude;
        this.longitude = longitude;
        this.distance = 0D;
        this.cardinal = "";
    }

    public void setImageVector(double distance, double direction, String cardinal) {
        this.distance = distance;
        this.direction = direction;
        this.cardinal = cardinal;
    }

    public File getFile() {
        return file;
    }

    public double getLatitude() {
        return latitude;
    }

    public double getLongitude() {
        return longitude;
    }

    public double getDistance() {
        return distance;
    }

    public double getDirection() {
        return direction;
    }

    public String getCardinal() {
        return cardinal;
    }
}

File: src/main/java/net/tlalka/imager/data/GeoImage.kt
package net.tlalka.imager.data

import java.io.File

class GeoImage(val file: File, val latitude: Double, val longitude: Double) {
    var distance: Double = 0.toDouble()
        private set
    var direction: Double = 0.toDouble()
        private set
    var cardinal: String = ""
        private set

    init {
        this.distance = 0.0
        this.cardinal = ""
    }

    fun setImageVector(distance: Double, direction: Double, cardinal: String) {
        this.distance = distance
        this.direction = direction
        this.cardinal = cardinal
    }
}
